src/library_gl.js

```diff
@@ -1081,31 +1097,20 @@ var LibraryGL = {
               }
             }
             return;
-          } else if (result instanceof WebGLBuffer ||
-                     result instanceof WebGLProgram ||
-                     result instanceof WebGLFramebuffer ||
-                     result instanceof WebGLRenderbuffer ||
-#if USE_WEBGL2
-                     result instanceof WebGLQuery ||
-                     result instanceof WebGLSampler ||
-                     result instanceof WebGLSync ||
-                     result instanceof WebGLTransformFeedback ||
-                     result instanceof WebGLVertexArrayObject ||
-#endif
-                     result instanceof WebGLTexture) {
-            ret = result.name | 0;
           } else {
```

Comment:
This kind of check causes isssues on browsers that only support WebGL1 - one would have to do `typeof WebGLQuery != undefined && result instanceof WebGLQuery` for all WebGL 2 types, which gets really verbose. Checking all these types makes little sense however, it is faster and much shorter to just get the field and deal with errors in an exception.